JSON JmesPath assertion

Ubik-Damien-Sailly opened the pull request:
## Description
Create a new assertion based on JmesPath http://jmespath.org/
https://github.com/burtcorp/jmespath-java

## Motivation and Context
JmesPath more powerful than JSON Path

## How Has This Been Tested?
JUnit
GUI

## Screenshots (if appropriate):

## Types of changes
- New feature (non-breaking change which adds functionality)


## Checklist:
- [x ] My code follows the [code style][style-guide] of this project.
- [ ] I have updated the documentation accordingly.

[style-guide]: https://wiki.apache.org/jmeter/CodeStyleGuidelines

Ubik-Damien-Sailly committed:
Add JmesPath assertion

Ubik-Damien-Sailly committed:
Code review: add Caffeine cache loader to JmesPathAssertion

    https://github.com/ben-manes/caffeine
    prevent compiling same JmesPath query several times

Ubik-Damien-Sailly committed:
Revert "Add JmesPath assertion"

This reverts commit 7cf0458d5582a39f7502a0fc06a2935a97d1b3bb.

Ubik-Damien-Sailly committed:
Add JmesPath assertion

ubikloadpack commented:
Peux-tu alimenter la documentation pour le composant ?
Voir component_reference.xml, la doc html est générée avec ant docs-site

Merci

Ubik-Damien-Sailly committed:
Add JmesPath documentation

Ubik-Damien-Sailly committed:
Code factorization for JSON GUI + documentation update

JSONPathAssertionGUI and JmesPathAssertionGUI are using same feature,
so to avoid code duplicate, setting JSONPath as JmesPath super class

ubikloadpack commented:
Contributed and Merged a bit differently through 
https://github.com/apache/jmeter/pull/498
